### DISCUSSION TITLE
Keycloak - Direct Reset Password URL

### DISCUSSION THREAD

**Discussion by eldarj** [keycloak/keycloak#14671] [Q&A]
I’d like to redirect a user from another app, directly to the Keycloak Password Reset flow, but unfortunately, there are some issues because of the missing `tab_id` query param.

This is the URL that I use

```
https://<keycloak>/auth/realms/<realm>/login-actions/reset-credentials?client_id=<client>
```

It works differently from when a user navigates to Keycloak’s login page and then clicks the Reset Password link. That generated URL by Keycloak contains a tab_id param, which for some reason makes some difference.
```
https://<keycloak>/auth/realms/<realm>/login-actions/reset-credentials?client_id=<client>&tab_id=<some-keycloak-generated-id>
```

I assume it’s how sessions are handled in the background. How can I make this work?

The goal is to just redirect from a Custom app to the Keycloak Password flow. Can I somehow generate a `tabId` myself if it doesn’t exist on Keycloak – do I need to initialize some kind of session and mimic what happens when a user opens up the login page?

Can this be done at all?

**Comment by pckhoi** [comment]
I'm not aware of a way to do that. The alternative is to simply send an email to your users with instructions to reset password, which is described here: https://lists.jboss.org/pipermail/keycloak-user/2018-October/015910.html

**Comment by DjordjeVitkovic** [comment]
Is there any solution on this?
I am trying to achieve something similar. I want tokens to be generated by another service, and that is possible by creating custom reset-credentials-email provider and that works. Then I cannot create link to reset password form directly...
I would like to open reset password page and I will add one more custom provider to check token by api call, and in the last, reset password will happened through Keycloak.
